static/js/compose.js

```diff
@@ -637,6 +643,39 @@ exports.initialize = function () {
         }
     });
 
+    // Show a warning if a private stream is linked
+    $(document).on('streamname_completed.zulip', function (event, data) {
+        if (compose_state.get_message_type() !== 'stream') {
+            return;
+        }
+
+        if (data !== undefined && data.stream !== undefined) {
```

Comment:
We should use early-return here.  I'll probably do that myself as a follow-up commit to demonstrate the technique.